spec/features/user_spec.rb

```diff
@@ -0,0 +1,94 @@
+  describe 'index page' do
+    before :each do
+      @user_first = User.create(name: 'Tom', photo: 'tom.png', bio: 'I am Tom', post_counter: 0)
+      @user_second = User.create(name: 'Lilly', photo: 'tom.png', bio: 'A beautiful lady', post_counter: 1)
+      @user_third = User.create(name: 'Ogaga', photo: 'tom.png', bio: 'A good guy', post_counter: 2)
+    end
+
+    it 'Should display all user names' do
+      visit '/'
+      expect(page).to have_content('Tom')
+      expect(page).to have_content('Lilly')
+      expect(page).to have_content('Ogaga')
+    end
+
+    it 'Should display the total post number of users' do
+      visit '/'
+      expect(page).to have_content('Number of posts: 0')
+      expect(page).to have_content('Number of posts: 1')
+      expect(page).to have_content('Number of posts: 2')
+    end
+
+    it "Should navigate to user's show page" do
+      visit user_path(@user_first.id)
+      expect(current_path).to eq(user_path(@user_first.id))
+    end
+  end
+
+  describe 'show page' do
+    before :each do
+      @user_first = User.create(name: 'Tom', photo: 'lilly.png', bio: 'A cute lady', post_counter: 0)
+
+      @post_first = @user_first.posts.new(title: 'Niklaus', text: 'This is my first post')
+      @post_first.likes_counter = 0
+      @post_first.comments_counter = 0
+      @post_first.update_users_posts_counter
+      @post_first.save
+
+      @post_second = @user_first.posts.new(title: 'Being a good child', text: 'This is my second post')
+      @post_second.likes_counter = 0
+      @post_second.comments_counter = 0
+      @post_second.update_users_posts_counter
+      @post_second.save
+
+      @post_third = @user_first.posts.new(title: 'Going to school', text: 'This is my third post')
+      @post_third.likes_counter = 0
+      @post_third.comments_counter = 0
+      @post_third.update_users_posts_counter
+      @post_third.save
+    end
+
+    it 'Should set the link of the picture to the img src' do
+      visit user_path(@user_first.id)
+      all('img').each do |i|
+        expect(i[:alt]).to eq('User Avatar')
+      end
+    end
+
+    it 'Should display the user name' do
+      visit user_path(@user_first.id)
+      expect(page).to have_content('Tom')
+    end
+
+    it 'Should display the total post number of 3' do
+      visit user_path(@user_first.id)
+      expect(page).to have_content('Number of posts: 3')
+    end
+
+    it 'Should display user bio' do
+      visit user_path(@user_first.id)
+      expect(page).to have_content('A cute lady')
+    end
+
+    it 'Should display the user first 3 posts' do
+      visit user_path(@user_first.id)
+      expect(page).to have_content('This is my first post')
+      expect(page).to have_content('This is my second post')
+      expect(page).to have_content('This is my third post')
+    end
+
+    it 'Should display a button with "See all posts" text' do
+      visit user_path(@user_first.id)
+      expect(page).to have_content('see all post')
+    end
```

Comment:
* Kindly write a test to cover this scenario `When I click a user's post, it redirects me to that post's show page` as its missing as its a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20a%20user%27s%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page)